packages/data/src/options/resolvers.js

```diff
@@ -19,7 +19,32 @@ export function* getOptionsWithRequest( names ) {
 		yield receiveOptions( results );
 		return results;
 	} catch ( error ) {
-		yield setRequestingError( error );
+		yield setRequestingError( error, names );
 		return error;
 	}
 }
+
+const fetches = {};
+
+export function* getOption( name ) {
+	yield setIsRequesting( name );
+	const names = yield getOptionsToRequest( name );
+
+	const fetchInProgress = fetches[ names ];
+	if ( fetchInProgress ) {
+		return;
+	}
+
+	const url = WC_ADMIN_NAMESPACE + '/options?options=' + names;
+	fetches[ names ] = true;
+	
+	try {
+		const result = yield apiFetch( { path: url } );
+		yield receiveOptions( result );
+	} catch ( error ) {
+		yield setRequestingError( error, names );
+	}
+
+	// Delete the fetch after to allow wp data to handle cache invalidation.
+	delete fetches[ names ];
```

Comment:
This comment is a bit ambiguous. Are you talking about `wp.data` handling _resolver_ cache invalidation here? If true, then no this won't handle resolver cache invalidation.

Reply:
Up for rewording this comment, but I believe it's accurate.  We need to delete the fetch from our local variable to **allow** (keyword) wp data to handle invalidation.  It does not invalidate cache directly, just allows it.

If we don't remove this fetch and we want to invalidate this cache using `invalidateResolution`, it would never actually work since the previous promise has already resolved.